app/controllers/api/v2/puppet_hosts_controller.rb

```diff
@@ -10,16 +10,15 @@ class PuppetHostsController < V2::BaseController
       param :id, :identifier_dottable, :required => true
 
       def puppetrun
-        return deny_access unless Setting[:puppetrun]
-        process_response @host.puppetrun!
+        render status: :not_implented
       end
 
       def action_permission
         case params[:action]
-          when 'puppetrun'
-            :puppetrun
-          else
-            super
+        when 'puppetrun'
+          :puppetrun
```

Comment:
haven't we dropped the permission?

I believe we can `prepend_before_action` and render there directly. Also, we can add deprecations for the `apidocs`.

Take a look at https://github.com/theforeman/foreman/pull/7800/files#diff-36882131819ddf97a1e444b073996328R5
I believe we can do the same here and keep just empty `puppetrun` method with the prepended filter, that will render the status.